lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -149,6 +174,77 @@ def remove_schedule(name)
 
       private
 
+      def supports_lua?
+        redis_master_version >= 2.5
+      end
+
+      def redis_master_version
+        Resque.redis.info['redis_version'].to_f
+      end
+
+      def startup_sha(refresh = false)
```

Comment:
Can you explain what `Resque.schedule=` is supposed to do at a high level? Why is it removing all the non-persisted schedule? What is a non-persisted schedule? What's a persisted schedule? This is all fairly internal to the repository vocab that I don't have.. what are they in the context of Shopify?